enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,352 @@
+  to create an object and to apply labels post cluster creation.
+
+### Non-Goals
+
+- Management(update/delete) of resource labels poallowed numberst creation of cluster is out of scope.
+
+## Proposal
+
+A label of the form `kubernetes-io-cluster-<cluster_id>:owned` will be added to every
+resource created by Openshift to enable administrator to differentiate the resources
+created for Openshift cluster. An administrator is not allowed to add or modify the label 
+having the prefix `kubernetes-io` or `openshift-io` in the name. The same can be found 
+applied to other cloud platform resources which supports labels/tags for ex: AWS, Azure.
+
+New `userLabels` field will be added to `platform.gcp` of install-config for the user 
+to define the labels to be added to the resources created by installer and in-cluster operators.
+
+If `platform.gcp.userLabels` in the install-config has any labels defined, then these labels 
+will be added to all the gcp resources created by OpenShift, provided all the defined 
+labels meet all the below conditions or else the cluster creation will fail.
+1. A label name must have minimum of 1 character and can have maximum of 63 characters.
+2. A label value can have maximum of 63 characters.
+3. A label name and value must contain only lowercase letters, numeric characters, 
+   underscores, and dashes.
+4. A label name must start with a lowercase letter.
+5. Each resource can have multiple labels, up to a maximum of 32.
+   - GCP supports a maximum of 64 labels per resource. Restricting the number of 
+    user defined labels to 32 and reserving 32 for Openshift's internal use.
+
+All in-cluster operators that create GCP resources (Cluster Infrastructure ,Storage ,Node
+,NetworkEdge, Internal Registry) will apply these labels during resource creation.
+
+The userLabels field is intended to be set at install time and is considered immutable. 
+Components that respect this field must only ever add labels that they retrieve from this 
+field to cloud resources, they must never remove labels from the existing underlying cloud 
+resource even if the labels are removed from this field(despite it being immutable).
+
+If the userLabels field is changed post-install, there is no guarantee about how an 
+in-cluster operator will respond to the change. Some operators may reconcile the 
+change and update the labels on the GCP resource. Some operators may ignore the change. 
+However, if labels are removed from userLabels, the label will not be removed from the 
+GCP resource.
+
+### Workflow Description
+
+- An Openshift administrator requests to add required labels to all GCP resources 
+  created by Openshift by adding it in `platform.gcp.userLabels`
+- Openshift installer validates the labels defined in `platform.gcp.userLabels` and 
+  adds these labels to all resources created during installation and also updates 
+  `.status.platformStatus.gcp.resourceLabels` of the `infrastructure.config.openshift.io`
```

Comment:
This field is on status for AWS, however, the AWS project has now been updated and they are adding the field to the spec, so that users can update the tags after cluster install should they so desire.

I think if we add this to the status, we will shoot ourselves in the foot since we will then also need to add the the spec later. I'm wondering if we should add this directly to spec and document that updates aren't supported on all components, until we can find a way to make sure they are.

CC @TrilokGeer as SME on the AWS tagging updates

Reply:
This comment can be resolved, will move forward with status

Reply:
Thank you for confirming.